client/app.jsx

```diff
@@ -60,7 +79,11 @@ export default class App extends React.Component {
     }
     if (route.path === 'details') {
       const businessId = route.params.get('businessId');
-      return <Details businessId={businessId} />;
+      let isSaved;
+      if (this.state.businessIds) {
+        isSaved = this.state.businessIds.includes(businessId);
+      }
```

Comment:
This is why i suggest using an empty array as that initial `businessIds` state. That way you can use 1 line of code instead of 4 because the `.includes()` will just return `false` if it's empty.